frontend/app/components/routing/wp-list/wp-list.controller.ts

```diff
@@ -98,51 +97,55 @@ function WorkPackagesListController($scope:any,
+      .take(1)
+      .subscribe(() => $scope.tableInformationLoaded = true);
 
-      updateTitle(query);
 
-      wpListChecksumService.updateIfDifferent(query, pagination as WorkPackageTablePagination);
-    });
+    // Update the title whenever the query
+    states.table.query.values$()
+      .takeUntil(scopeDestroyed$($scope))
+      .subscribe((query) => updateTitle(query));
 
-    wpTablePagination.observeOnScope($scope)
-      .withLatestFrom(scopedObservable($scope, states.table.query.values$()))
+    states.table.context.fireOnStateChange(wpTablePagination.state, 'Query loaded')
+      .values$()
+      .withLatestFrom(states.table.query.values$())
+      .takeUntil(scopeDestroyed$($scope))
       .subscribe(([pagination, query]) => {
-      if (wpListChecksumService.isQueryOutdated(query, pagination as WorkPackageTablePagination)) {
-        wpListChecksumService.update(query, pagination as WorkPackageTablePagination);
+        updateTitle(query);
+        if (wpListChecksumService.isQueryOutdated(query, pagination)) {
+          wpListChecksumService.update(query, pagination);
 
-        updateResultsVisibly();
-      }
+          updateResultsVisibly();
+        }
     });
 
     setupChangeObserver(wpTableFilters);
     setupChangeObserver(wpTableGroupBy);
     setupChangeObserver(wpTableSortBy);
     setupChangeObserver(wpTableSum);
-    setupChangeObserver(wpTableTimeline, false);
-    setupChangeObserver(wpTableTimeline, false);
-    setupChangeObserver(wpTableHierarchies, false);
-    setupChangeObserver(wpTableColumns, false);
+    setupChangeObserver(wpTableTimeline);
+    setupChangeObserver(wpTableTimeline);
+    setupChangeObserver(wpTableHierarchies);
+    setupChangeObserver(wpTableColumns);
   }
 
-  function setupChangeObserver(service:WorkPackageTableBaseService, triggerUpdate:boolean = true) {
+  function setupChangeObserver(service:WorkPackageQueryStateService) {
     const queryState = states.table.query;
 
     states.table.context.fireOnStateChange(service.state, 'Query loaded')
       .values$()
       .takeUntil(scopeDestroyed$($scope))
-      .filter(() => !isAnyDependentStateClear()) // Avoid updating while not all states are initialized
-      .filter(() => queryState.hasValue())
-      .filter((stateValue:WorkPackageTableQueryState) => stateValue.hasChanged(queryState.value!))
-      .subscribe((stateValue:WorkPackageTableQueryState) => {
+      .filter(() => queryState.hasValue() && service.hasChanged(queryState.value!))
+      .subscribe(() => {
         const newQuery = queryState.value!;
-        stateValue.applyToQuery(newQuery);
+        const triggerUpdate = service.applyToQuery(newQuery);
         states.table.query.putValue(newQuery);
 
+        // Update the current checksum
+        wpListChecksumService.updateIfDifferent(newQuery, wpTablePagination.current);
+
+        // Update the page, if the change requires it
         if (triggerUpdate) {
           updateResultsVisibly(true);
```

Comment:
The whole of the table reloads, including backend requests, when moving columns and I believe this change is the reason for it.